src/amo/reducers/home.js

```diff
@@ -310,6 +312,7 @@ export const createInternalShelf = (
     title: selectLocalizedContent(result.title, lang),
     url: result.url,
     endpoint: result.endpoint,
+    addonType: result.addonType || 0,
```

Comment:
You shouldn't need the `|| 0` here, as without it `addonType` will end up with a `null` which should work. Flow might complain and if it does you'll need to change the type above to `AddonTypeType | null`.